From 94555facc1d72d70fc6b297d867662749d894731 Mon Sep 17 00:00:00 2001
From: Aaron <blue@aquarat.co.uk>
Date: Sat, 5 Sep 2026 22:18:37 +0100
Subject: [PATCH] cmake: drop -D__STRICT_ANSI__ on Linux to fix the build with
 GCC 16

With GCC 16's libstdc++ the Linux build fails when __STRICT_ANSI__ is
defined (__int128 redefinition), so stop adding the define to
CMAKE_CXX_FLAGS on Linux.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33930c95..a055d447 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -385,7 +385,7 @@ if (NOT MSVC)
 endif()
 
 if(CMAKE_SYSTEM_NAME MATCHES "^Linux")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__STRICT_ANSI__")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ")
     if(CMAKE_SYSTEM_PROCESSOR MATCHES "^armv7")
         add_definitions(-mfpu=neon)    #please define in project/cross-compile/arm.toolchain.cmake
     endif()
-- 
2.55.0

